src/main/java/com/conveyal/datatools/manager/models/transform/NormalizeFieldTransformation.java

```diff
@@ -0,0 +1,354 @@
+    private static final String SUBSTITUTION_SEPARATOR = "=>";
+
+    /** The field name in the GTFS table being transformed. */
+    public String fieldName;
+
+    /** Whether to perform capitalizations. */
+    public boolean capitalize = true;
+
+    /** Whether to perform text substitutions. */
+    public boolean performSubstitutions = true;
+
+    /**
+     * Capitalization exceptions are initialized with the default configured ones,
+     * and can be overridden from the UI.
+     */
+    public String capitalizeExceptions = DEFAULT_EXCEPTIONS;
+
+    /**
+     * Substitutions are initialized with the default configured ones,
+     * and can be overridden from the UI.
+     */
+    public String substitutions = DEFAULT_SUBSTITUTIONS;
+
+    // These fields are reset when the setter is called
+    // and initialized when executing the transform method.
+    private List<Substitution> capitalizeSubstitutions;
+    private List<Substitution> stringSubstitutions;
+
+    private List<Substitution> getCapitalizeSubstitutions() {
+        if (capitalizeSubstitutions == null) {
+            initializeCapitalizeSubstitutions();
+        }
+        return capitalizeSubstitutions;
+    }
+
+    private List<Substitution> getStringSubstitutions() {
+        if (stringSubstitutions == null) {
+            initializeStringSubstitutions();
+        }
+        return stringSubstitutions;
+    }
+
+    /**
+     * Used in tests to create an instance of this transform.
+     * @param table The table with the field to modify.
+     * @param fieldName The field in the specified table to modify.
+     * @param exceptions A list of capitalization exceptions:
+     *                   - if null, the default configured exceptions will be used.
+     *                   - if empty, no capitalization exceptions will be considered.
+     * @param substitutions A list of substitutions:
+     *                      - if null, the default configured substitutions will be performed.
+     *                      - if empty, no substitutions will be performed.
+     * @return An instance of the transformation with the desired settings.
+     */
+    public static NormalizeFieldTransformation create(
+        String table, String fieldName, String exceptions, String substitutions)
+    {
+        NormalizeFieldTransformation transformation = new NormalizeFieldTransformation();
+        transformation.fieldName = fieldName;
+        transformation.table = table;
+        // Override configured defaults if corresponding args are not null.
+        // (To set up no capitalization exceptions or no substitutions, pass an empty string.)
+        if (exceptions != null) {
+            transformation.capitalizeExceptions = exceptions;
+        }
+        if (substitutions != null) {
+            transformation.substitutions = substitutions;
+        }
+        return transformation;
+    }
+    public static NormalizeFieldTransformation create(String table, String fieldName) {
```

Comment:
Can you add a line break before this line?

Reply:
Updated in https://github.com/ibi-group/datatools-server/pull/371/commits/cd883d91e7d11fa2c698675980d8443f6869943f.